src/golang/lib/response/node.go

```diff
@@ -11,6 +11,85 @@ import (
 
 // This file should map exactly to
 // `src/ui/common/src/handlers/responses/node.ts`
+type MergedNode struct {
```

Comment:
I feel merged node is not very descriptive by staring at the name. I'd personally prefer something longer but more clear, like `OperatorWithArtifact` or `MergedOperatorArtifact`.